pubsub/pulsar/pulsar.go

```diff
@@ -82,22 +83,28 @@ func (p *Pulsar) Init(metadata pubsub.Metadata) error {
 }
 
 func (p *Pulsar) Publish(req *pubsub.PublishRequest) error {
-	producer, err := p.client.CreateProducer(pulsar.ProducerOptions{
-		Topic: req.Topic,
-	})
-	if err != nil {
-		return err
+	if p.producer == nil || p.producer.Topic() != req.Topic {
```

Comment:
This will be helpful but I think we should take it a step further and have a multiplexer that will know which producer to use based on the topic and close them all in the end.

This could also be achieved with some sort of LRU cache, as long as the producers are closed() before being deleted from cache.

Reply:
Updated PR for this, please review.